src/App.tsx

```diff
@@ -1,24 +1,221 @@
+  );
+
+  const selectAllTodos = useCallback(async () => {
+    try {
+      await Promise.all(todos.map(todo => (
+        onUpdate(todo.id))));
+
+      setTodos(todos.map(todo => (
+        { ...todo, completed: !isAllTodosCompleted }
+      )));
+    } catch {
+      setErrorNotification(ErrorNotification.UPDATE);
+    }
+  }, [todos]);
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
+  const addTodo = async () => {
+    const newTodo: RequestTodos = {
+      userId: USER_ID,
+      completed: false,
+      title: search,
+    };
+
+    setTempTodo({
+      id: 0,
+      ...newTodo,
+    });
+
+    createTodo(USER_ID, newTodo)
+      .then(res => setTodos(prevTodos => [...prevTodos, res]))
+      .catch(() => setErrorNotification(ErrorNotification.ADD))
+      .finally(() => setTempTodo(null));
+
+    setSearch('');
+  };
+
+  const handleDeleteTodo = async (id: number) => {
+    setTempTodo({
+      id,
+      userId: 0,
+      title: '',
+      completed: false,
+    });
+
+    deleteTodo(id)
+      .then(() => setTodos(prevTodos => prevTodos
+        .filter(todo => todo.id !== id)))
+      .catch(() => setErrorNotification(ErrorNotification.DELETE))
+      .finally(() => setTempTodo(null));
+  };
+
+  const deleteTodosCompleted = async (idsForDeleting: number[]) => {
+    setTodosForDeleting([...idsForDeleting]);
+    todos.filter(todo => todo.completed)
+      .map(todo => handleDeleteTodo(todo.id));
+  };
+
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-add-and-delete#react-todo-app-add-and-delete">React Todo App - Add and Delete</a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <Header
+          todos={todos}
+          search={search}
+          setSearch={setSearch}
+          addTodo={addTodo}
+          setErrorNotification={setErrorNotification}
+          selectAllTodos={selectAllTodos}
+        />
+
+        {todos.length > 0 && (
+          <>
+            <TodoList
+              todos={initialTodos}
+              handleDeleteTodo={handleDeleteTodo}
+              tempTodo={tempTodo}
+              todosForDeleting={todosForDeleting}
+              changeStatus={updateTodoStatus}
+            />
+            <Footer
+              todos={initialTodos}
+              todoFilter={todoFilter}
+              setTodoFilter={setTodoFilter}
+              deleteTodosCompleted={deleteTodosCompleted}
+              completedId={completedID}
+            />
+          </>
+        )}
+      </div>
+
+      {errorNotification && (
+        <TodoError
+          errorNotification={errorNotification}
+          closeError={() => setErrorNotification(ErrorNotification.NONE)}
```

Comment:
Move this logic into TodoError component
```suggestion
          closeError={setErrorNotification}
```